### PR TITLE
[fix #128] Delete also parent process which is not sucessfully deleted

### LINKED ISSUE #128: Wildfly containers don't shutdown properly on Mac OSX
After wildfly container shuts down, I still have bash process running which consumes ~50% cpu. It happens only with wildfly.

### PR DESCRIPTION
**Pull request by llowinge**

### CODE REVIEW

File: core/src/main/java/org/jboss/qa/jcontainer/util/ProcessUtils.java [hunk 1]
```diff
@@ -65,8 +87,17 @@ public static String getJavaPidByContainerId(long id) {
 		}
 	}
 
+	private static String getParentPidOfPid(String pid) {
+		final String command = String.format("ps -ajx | awk '$2 == \"%s\"' | awk '{print $3}'", pid);
```
[the remark this answers is not in the document]
Comment (llowinge) [reply core/src/main/java/org/jboss/qa/jcontainer/util/ProcessUtils.java]: Fixed

File: core/src/main/java/org/jboss/qa/jcontainer/util/ProcessUtils.java [hunk 2]
```diff
@@ -47,6 +47,28 @@ public static void killJavaByName(String processName) {
 		}
 	}
 
+	public static void killAllJavaContainerProcesses(long id) {
```
[the remark this answers is not in the document]
Comment (llowinge) [reply core/src/main/java/org/jboss/qa/jcontainer/util/ProcessUtils.java]: Yep

File: core/src/test/java/org/jboss/qa/jcontainer/util/executor/SubProcessKillTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+@Slf4j
+public class SubProcessKillTest {
+
+	private static final int CONTAINER_ID = 1234;
+
+	private static List<String> commands;
+
+	@BeforeClass
+	public static void setUp() {
+		Assume.assumeTrue(SystemUtils.IS_OS_MAC);
```
Comment (osmman) [inline core/src/test/java/org/jboss/qa/jcontainer/util/executor/SubProcessKillTest.java]: Let this test to be executed on all UNIX platforms and exclude Windows

File: core/src/test/java/org/jboss/qa/jcontainer/util/executor/SubProcessKillTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+				}
+			}
+		});
+		final String javaPid = ProcessUtils.getJavaPidByContainerId(CONTAINER_ID);
+		Assert.assertTrue("Java PID should be a number", StringUtils.isNumeric(javaPid));
+
+		final String parentPid = ProcessUtils.getParentPidOfPid(javaPid);
+		Assert.assertTrue("Parent PID should be a number", StringUtils.isNumeric(parentPid));
+
+		ProcessUtils.killAllJavaContainerProcesses(CONTAINER_ID);
```
Comment (osmman) [inline core/src/test/java/org/jboss/qa/jcontainer/util/executor/SubProcessKillTest.java]: add assert which check that all processes were killed

Comment (llowinge) [reply core/src/test/java/org/jboss/qa/jcontainer/util/executor/SubProcessKillTest.java]: Already checked with ```Assert.assertEquals("Response code should be 137=128+9 SIGKILL", 137, process.waitFor());```

**Pull request merged**